### PR TITLE
Add tenant slug and tests; closes #72

### LINKED ISSUE #72: parameterize tenant organization names and add tenant unit tests

### PR DESCRIPTION
**Pull request by EmilyMB**
This involves general cleanup and organization. Rake db:migrate after this PR is merged.

Changes include:
* Add slug column to the tenant model (for increase clarity and url validation)
* Remove/add misspelled column on item table
* Fixed controllers/tenants/items_controller to reroute unknown/misspelled tenant slug's to root
* Added unit tests for tenant model.

### CODE REVIEW

File: app/controllers/tenants/items_controller.rb [hunk 1]
```diff
@@ -31,6 +33,7 @@ def update
   private
 
   def item_params
-    params.require(:item).permit(:title, :price, :description, :image)
+    params.require(:item).permit(:title, :price, :description, :retired,
+                                 :tenant_id, :repayment_begin, :repayment_rate)
```
Comment (mikedao) [inline app/controllers/tenants/items_controller.rb]: I love strong params. SO much.

:sparkles:

File: app/models/tenant.rb [hunk 2]
```diff
@@ -2,4 +2,13 @@ class Tenant < ActiveRecord::Base
...
   has_many :categories, through: :items
   has_many :user
+
+  before_save :create_slug
+
+  validates :location,     presence: true
+  validates :organization, presence: true
+
+  def create_slug
+    self.slug = organization.parameterize
```
Comment (mikedao) [inline app/models/tenant.rb]: Just like we agreed on!

File: db/migrate/20150213025747_updated_items.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class UpdatedItems < ActiveRecord::Migration
+  def change
+    remove_column :items, :reqpayment_begin
```
Comment (mikedao) [inline db/migrate/20150213025747_updated_items.rb]: This is my fault. I should have been more careful.

File: test/integration/guest_user_test.rb [hunk 4]
```diff
@@ -77,13 +77,14 @@ class GuestUserTest < ActionDispatch::IntegrationTest
     non_tenant_category = create(:category, name: "bad cats")
     tenant_category = create(:category, name: "agriculture")
     non_tenant_item = create(:item, categories: [non_tenant_category])
-    tenant = create(:tenant, organization: "bob")
+    tenant = create(:tenant, organization: "bob's")
```
Comment (mikedao) [inline test/integration/guest_user_test.rb]: Why the change here?

File: test/integration/guest_user_test.rb [hunk 5]
```diff
@@ -92,6 +93,12 @@ class GuestUserTest < ActionDispatch::IntegrationTest
     refute page.has_content?(non_tenant_item.title)
   end
 
+  test "will be redirected to home page if tenant does not exist" do
+    visit tenant_path(slug: "made-up-shop")
+
```
Comment (mikedao) [inline test/integration/guest_user_test.rb]: Have we decided that all errors revert to root_path?

**Comment by mikedao** [comment]
I didn't want to leave an inline comment for this, but the tests look good.

File: test/integration/guest_user_test.rb (hunk 4 above)
Comment (EmilyMB) [inline test/integration/guest_user_test.rb]: I wanted to ensure that the organization was being parameterized correctly

File: test/integration/guest_user_test.rb (hunk 5 above)
Comment (EmilyMB) [inline test/integration/guest_user_test.rb]: No, we haven't talked about that. I was told that having all incorrect routes rerouted to root using a splat operator is sloppy. We can ask for guidance tomorrow!

File: db/migrate/20150213025747_updated_items.rb (hunk 3 above)
Comment (EmilyMB) [inline db/migrate/20150213025747_updated_items.rb]: np! We were all there, so many :eyes:, and we still missed it :laughing:

**Comment by mikedao** [comment]
This all looks great!
 :sparkles:  :sparkles: :sparkles: :sparkles:
